hail_search/queries/mito.py

```diff
@@ -118,6 +121,179 @@ class MitoHailTableQuery(BaseHailTableQuery):
+    ) -> tuple[hl.Table, hl.Table]:
+        sample_types = set(project_sample_type_data.keys())
+        if len(sample_types) == 1:
+            # Import and filter entry table for one family or one project with one sample type
+            return super()._import_and_filter_entries_ht(project_guid, num_families, project_sample_type_data, **kwargs)
+
+        self._has_both_sample_types = True
+        entries = {}
+        for sample_type in sample_types:
+            ht, sample_data = self._load_family_or_project_ht(
+                num_families, project_guid, project_sample_type_data, sample_type, **kwargs
+            )
+            entries[sample_type] = (ht, sample_data)
+
+        return self._filter_entries_ht_both_sample_types(
+            *entries[SampleType.WES.value], *entries[SampleType.WGS.value], **kwargs
+        )
+
+    def _import_and_filter_multiple_project_hts(
+        self, project_samples: dict, n_partitions=MAX_PARTITIONS, **kwargs
+    ) -> tuple[hl.Table, hl.Table]:
+        single_sample_type_project_samples = {}
+        both_sample_type_project_samples = {}
+        for project_guid, sample_dict in project_samples.items():
+            if len(sample_dict) == 1:
+                single_sample_type_project_samples[project_guid] = sample_dict
+            else:
+                both_sample_type_project_samples[project_guid] = sample_dict
+
+        filtered_project_hts = []
+        filtered_comp_het_project_hts = []
+
+        # Process projects with only one sample type separately
+        if single_sample_type_project_samples:
+            ht, ch_ht = super()._import_and_filter_multiple_project_hts(single_sample_type_project_samples, n_partitions, **kwargs)
+            if ht is not None:
+                filtered_project_hts.append(ht)
+            if ch_ht is not None:
+                filtered_comp_het_project_hts.append(ch_ht)
+
+        if both_sample_type_project_samples:
+            self._has_both_sample_types = True
+            entries = self._load_project_hts_both_sample_types(both_sample_type_project_samples, n_partitions, **kwargs)
+            for entry in entries:
+                wes_ht, wes_project_samples = entry[SampleType.WES.value]
+                wgs_ht, wgs_project_samples = entry[SampleType.WGS.value]
+                ht, comp_het_ht = self._filter_entries_ht_both_sample_types(
+                    wes_ht, wes_project_samples, wgs_ht, wgs_project_samples,
+                    is_merged_ht=True, **kwargs
+                )
+                if ht is not None:
+                    filtered_project_hts.append(ht)
+                if comp_het_ht is not None:
+                    filtered_comp_het_project_hts.append(comp_het_ht)
+
+        return self._merge_filtered_hts(filtered_comp_het_project_hts, filtered_project_hts, n_partitions)
+
+    def _load_project_hts_both_sample_types(self, project_samples, n_partitions, **kwargs):
+        def initialize_project_hts():
```

Comment:
I can't tell if this is an antipattern. I didn't want to make these class-level functions because I wanted them to be called only when _load_project_hts_both_sample_types on the mito class is called, not from [this](https://github.com/broadinstitute/seqr/pull/4330/files#diff-71932dea9e59d8d1ce70d88444658acffce2f3865c9afb11b57f93daf6610f66R160) super() call.

Reply:
Alternatively, maybe you could leave the data structures alone and just call `_load_project_hts` once per sample type? There might be an edge case I am missing, but if not I think it would be relatively straightforward to change `_load_project_hts ` to return the hts and the sample data and then just do something like this in `_import_and_filter_multiple_project_hts`:

```
self._has_both_sample_types = True
wes_entries =  self._load_project_hts(
    both_sample_type_project_samples[SampleType.WES.value], n_partitions, **kwargs,
)
wgs_entries = self._load_project_hts(
    both_sample_type_project_samples[SampleType.WGS.value], n_partitions, **kwargs,
)
for i, (wes_ht, wes_project_samples) in enumerate(wes_entries):
    wgs_ht, wgs_project_samples = wgs_entries[i]
    ...
```

Reply:
ok, I made a few changes to simply this section and added some post-processing in the mito `_load_project_hts`

Reply:
oh I just saw your second comment that is even better